tests/core/test_stellargraph.py

```diff
@@ -427,3 +429,59 @@ def test_feature_conversion_from_iterator():
     ab = gs.get_feature_for_nodes([4, 5], "B")
     assert ab.shape == (2, 10)
     assert ab[:, 0] == pytest.approx([4, 5])
+
+
+def example_benchmark_graph(
+    feature_size=None, n_nodes=100, n_edges=200, n_types=4, features_in_nodes=True
+):
+    G = nx.Graph()
+
+    G.add_nodes_from(range(n_nodes))
+    edges = [
+        (random.randint(0, n_nodes - 1), random.randint(0, n_nodes - 1))
+        for _ in range(n_edges)
+    ]
+    G.add_edges_from(edges)
+
+    for v in G.nodes():
+        G.nodes[v]["label"] = v % n_types
+
+    # Add example features
+    if feature_size is None:
+        node_features = None
+    elif features_in_nodes:
+        node_features = "feature"
+        for v in G.nodes():
+            G.nodes[v][node_features] = np.ones(feature_size)
+    else:
+        node_features = {}
+        for ty in range(n_types):
+            type_nodes = range(ty, n_nodes, n_types)
+            if len(type_nodes) > 0:
+                node_features[ty] = pd.DataFrame(
+                    [np.ones(feature_size)] * len(type_nodes), index=type_nodes
+                )
+
+    return G, node_features
+
+
+@pytest.mark.benchmark(group="StellarGraph creation", timer=snapshot)
+# various element counts, to give an indication of the relationship
+# between those and memory use (0,0 gives the overhead of the
+# StellarGraph object itself, without any data)
+@pytest.mark.parametrize("num_nodes,num_edges", [(0, 0), (100, 0), (100, 200)])
+# various feature sizes (including no features) to capture that cost
+@pytest.mark.parametrize("feature_size", [None, 1, 100])
+# test both features
+@pytest.mark.parametrize("features_in_nodes", [False, True])
```

Comment:
Do you have an alternative? I suspect that the `features_in_node` option will become irrelevant and thus totally disappear with a non-networkx implementation, but for now it's useful as a baseline? But... maybe that's irrelevant anyway, and we can still remove?

That is, remove this and add homogeneity

```suggestion
@pytest.mark.parametrize("num_types", [1, 4])
```

or, keep it and add homogeneity?

```suggestion
@pytest.mark.parametrize("features_in_nodes", [False, True])
@pytest.mark.parametrize("num_types", [1, 4])
```

Reply:
I would tend to remove it: as you say, it will be irrelevant soon (hopefully!)

Reply:
Sounds good!

Reply:
I've switched to:

- sizes: (0, 0), (100, 200), (1000, 5000)
- feature size: `None`, `100`

I've retained the (0, 0) because I think it's helpful to know the "base" level for these sort of benchmarks, to know the lower bound for improvements, and to know how much an improvement actually changes the thing being edited (e.g. a change to vertices that takes the overall size from 100k to 90k might still be a 6x improvement in the size of vertices if they went from 12k to 2k). This is also why I had benchmarks with no edges.

I could be convinced otherwise, if we decide that's unlikely to be interesting.

Reply:
True, there will always be a base level to consider.  I am happy with these 👍